text/0517-io-os-reform.md

```diff
@@ -483,7 +483,72 @@ throughout IO, we can go on to explore the modules in detail.
 ### `std::env`
 [std::env]: #stdenv
 
-> To be added in a follow-up PR.
+Most of what's available in `std::os` today will move to `std::env`,
+and the signatures will be updated to follow this RFC's
+[Design principles] as follows.
+
+**Arguments**:
+
+* `args`: change to yield an iterator rather than vector if possible; in any case, it should produce an `OsStrBuf`.
```

Comment:
The API we're supporting is calling a free function and basically plucking out the arguments from somewhere in memory. On Windows and OSX this form of API is supported by the system (there are APIs to deal with this), but on linux the only source for arguments is the initial `main` function. This means that on linux we have specific code [to cache arguments oureslves](https://github.com/rust-lang/rust/blob/master/src/libstd/rt/args.rs#L42-L137).

If we did not squirrel away the arguments ourselves I don't think there's another vector to learn about the process's arguments at runtime from an arbitrary point (but I could be wrong!)